lib/node_modules/@stdlib/array/empty-like/docs/types/index.d.ts

```diff
@@ -0,0 +1,798 @@
+* @param dtype - data type
+* @returns empty array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'uint8' );
+* // returns <Uint8Array>
+*/
+declare function emptyLike( x: AnyArray, dtype: 'uint8' ): Uint8Array;
+
+/**
+* Creates an uninitialized array having the same length as a provided input array.
+*
+* ## Notes
+*
+* -   In browser environments, the function always returns zero-filled arrays.
+* -   In Node.js versions `>=3.0.0`, the underlying memory of returned typed arrays is **not** initialized. Memory contents are unknown and may contain **sensitive** data.
+*
+* @param x - input array from which to derive the output array length
+* @param dtype - data type
+* @returns empty array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'uint8c' );
+* // returns <Uint8ClampedArray>
+*/
+declare function emptyLike( x: AnyArray, dtype: 'uint8c' ): Uint8ClampedArray;
+
+/**
+* Creates a zero-filled array having a specified length.
+*
+* @param x - input array from which to derive the output array length
+* @param dtype - data type
+* @returns zero-filled array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'generic' );
+* // returns <number[]>
+*/
+declare function emptyLike( x: AnyArray, dtype: 'generic' ): Array<number>;
+
+/**
+* Creates an uninitialized array having the same length as a provided input array.
+*
+* The function supports the following data types:
+*
+* -   `float64`: double-precision floating-point numbers (IEEE 754)
+* -   `float32`: single-precision floating-point numbers (IEEE 754)
+* -   `complex128`: double-precision complex floating-point numbers
+* -   `complex64`: single-precision complex floating-point numbers
+* -   `int32`: 32-bit two's complement signed integers
+* -   `uint32`: 32-bit unsigned integers
+* -   `int16`: 16-bit two's complement signed integers
+* -   `uint16`: 16-bit unsigned integers
+* -   `int8`: 8-bit two's complement signed integers
+* -   `uint8`: 8-bit unsigned integers
+* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
+* -   `generic`: generic JavaScript values
+*
+* ## Notes
+*
+* -   In browser environments, the function always returns zero-filled arrays.
+* -   In Node.js versions `>=3.0.0`, the underlying memory of returned typed arrays is **not** initialized. Memory contents are unknown and may contain **sensitive** data.
+*
+* @param x - input array from which to derive the output array length
+* @param dtype - data type
+* @returns empty array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
```

Comment:
The examples should match the signature. In this case, `x` should be a `Float64Array`. Similarly, if `x: Int32Array`, then the example should do `var x = zeros( 2, 'int32' )`.

In short, examples should agree with the signature; otherwise, users are likely to get confused when seeing docs in their IDE.